spec/index.adoc

```diff
@@ -67,12 +67,12 @@ To get some consistency in this document, we try to use the following terms with
 
 We avoid the term “client”, as it could be the client of the Internet Computer or the client inside the distributed network that makes up the Internet Computer. Instead, we use the term _user_ to denote the external entity interacting with the Internet Computer, even if in most cases it will be some code (sometimes called “agent”) acting on behalf of a (human) user.
 
-The public entry points of canisters are called _methods_. Methods can be declared to be either _update methods_ (state mutation is preserved) or _query methods_ (state mutation is discarded, no further calls can be made).
+The public entry points of canisters are called _methods_. Methods can be declared to be either _update methods_ (state mutation is preserved), _query methods_ (state mutation is discarded, no further calls can be made), or _composite query_ methods (state mutation is discarded, calls can be made to canisters on the same subnet).
 
 Methods can be _called_, from _caller_ to _callee_, and will eventually incur a _response_ which is either a _reply_ or a _reject_. A method may have _parameters_, which are provided with concrete _arguments_ in a method call.
 
-External calls can be update calls, which can call both kinds of methods, and query calls, which can _only_ call query methods.
-Inter-canister calls can also call both kinds of methods. Note that calls from a canister to itself also count as "inter-canister".
+External calls can be update calls, which can call update and query methods, and query calls, which can _only_ call query and composite query methods.
+Inter-canister calls can call update and query methods (just like update calls). Note that calls from a canister to itself also count as "inter-canister".
```

Comment:
This is a complete list of calls that are allowed.

update -> {update, query}
query -> {}
composite query -> {query, composite query}

Should we specify this in a bit more detail?

Reply:
Now I've specified this a few lines above:
```
The public entry points of canisters are called _methods_. Methods can be declared to be either _update methods_ (state mutation is preserved, can call update and query methods of arbitrary canisters), _query methods_ (state mutation is discarded, no further calls can be made), or _composite query_ methods (state mutation is discarded, can call query and composite query methods of canisters on the same subnet).
```